kobo/apps/constance_backends/utils.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import annotations
+
+import json
+
+from typing import Union
+from kpi.utils.json import LazyJSONSerializable
+
+
+def to_python_object(
+    object_or_str: Union['LazyJSONSerializable', str]
+) -> Union[list, dict]:
+    """
+    Returns a python object from `object_or_str`.
+    """
+    # If a Constance setting is a `LazyJSONSerializable` object,
+    # `constance.config.SETTING` does not return the same type of object depending
+    # on whether the setting was saved in its back end or not. From the back end,
+    # it returns the string representation. Otherwise, it returns the
+    # `LazyJSONSerializable` object itself.
```

Comment:
Short answer: yes.


Long answer with more details to what I tried to explain in my comments. 

Actually, it is saved in [PostgreSQL then cached in Redis](https://github.com/kobotoolbox/kpi/blob/1c8a08fae6a2cff20019d08c56197d478bf1b31e/kobo/settings/base.py#L516-L518) after the first access of the variable. 
 
So when we call `constance.config.MY_ENV`, `constance`:
- gets [the default value](https://github.com/jazzband/django-constance/blob/master/constance/base.py#L16)
- try to get a [value from the back end](https://github.com/jazzband/django-constance/blob/master/constance/base.py#L19-L20), but if it does not exist, falls back on the default value (i.e. `LazyJSONSerializable` object).

After the first call of `constance.config.MY_ENV`, `constance` saves a serialized version  of `MY_ENV` in the back end and cached it Django CACHE (if enabled obviously). So in production, we are always getting a `str` even if its default has not been overridden.  In unit tests, PostgreSQL is always empty because tables are wiped after each test and Django CACHE is disabled.

Reply:
Thanks, very educational! I don't think we're going to explain all this in the comment (unless you think we should :thinking:) and current comment as-is leaves me wanting more detail to understand what it means.

Maybe we should write a simpler comment such as:
> Make `constance.config.SETTING` return a consistent value when `SETTING` is a `LazyJSONSerializable` object

Reply:
No, straight to the point ;-) Love it